sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -47,7 +46,7 @@ case class CachedData(plan: LogicalPlan, cachedRepresentation: InMemoryRelation)
 class CacheManager extends Logging {
 
   @transient
-  private val cachedData = new java.util.LinkedList[CachedData]
+  private var cachedData = IndexedSeq[CachedData]()
```

Comment:
I'm not 100 sure though, we don't need `volatile` for this var?

Reply:
Yes, I read through this and we either need to use the read locks or use @volatile here.  I am not an expert on the details here but @volatile seems like a better solution.  I think ultimately the differences are negligible.